src/modules/compliance/src/lib/Evaluator.cpp

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <string>
+#include <memory>
+#include <map>
+#include <sstream>
+
+#include <errno.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <stdarg.h>
+#include <ctype.h>
+#include <parson.h>
+#include <CommonUtils.h>
+#include <UserUtils.h>
+#include <SshUtils.h>
+#include <Logging.h>
+
+
+#include "Evaluator.h"
+#include "Result.h"
+
+namespace compliance {
+
+Result<bool> Evaluator::ExecuteAudit(char** payload, int* payloadSizeBytes)
+{
+    if (mJson == nullptr)
+    {
+        OsConfigLogError(mLog, "Invalid argument: rule is null");
+        return Error("Rule is null");
+    }
+
+    if(nullptr == payload || nullptr == payloadSizeBytes)
```

Comment:
>if(nullptr == payload || nullptr == payloadSizeBytes) [](http://example.com/codeflow?start=4&length=53)

One general rule that we try to respect in this project is to not rely on just operator precedence. Always add parenthesis.  For example, here:

'
if((nullptr == payload) || (nullptr == payloadSizeBytes))
'

Also for this particular line here, may also want to check for *payloadSizeBytes being <= 0.